tests/func/test_data_status.py

```diff
@@ -0,0 +1,148 @@
+from os.path import join
+
+from dvc.repo import Repo
+from dvc.utils.fs import remove
+
+EMPTY_STATUS = {
+    "committed": {},
+    "uncommitted": {},
+    "git": {},
+    "not_in_cache": [],
+    "unchanged": [],
+    "untracked": [],
+}
+
+
+def test_file(M, tmp_dir, scm, dvc):
+    tmp_dir.dvc_gen("foo", "foo", commit="add foo")
+    tmp_dir.dvc_gen("foo", "foobar")
+    remove(tmp_dir / "foo")
+
+    expected = {
+        "committed": {"modified": ["foo"]},
+        "uncommitted": {"deleted": ["foo"]},
+        "git": M.instance_of(dict),
+        "not_in_cache": [],
+        "unchanged": [],
+        "untracked": [],
+    }
+    assert dvc.data_status() == expected
+    assert dvc.data_status(granular=True) == expected
+
+
+def test_directory(M, tmp_dir, scm, dvc):
+    tmp_dir.dvc_gen({"dir": {"foo": "foo"}}, commit="add dir")
+    tmp_dir.dvc_gen({"dir": {"foo": "foo", "bar": "bar", "foobar": "foobar"}})
+    remove(tmp_dir / "dir")
+    (tmp_dir / "dir").gen({"foo": "foo", "bar": "barr", "baz": "baz"})
+    tmp_dir.gen("untracked", "untracked")
+
+    assert dvc.data_status() == {
+        "committed": {"modified": [join("dir", "")]},
+        "uncommitted": {"modified": [join("dir", "")]},
+        "git": M.instance_of(dict),
+        "not_in_cache": [],
+        "unchanged": [],
+        "untracked": [],
+    }
+
+    assert dvc.data_status(granular=True, untracked_files="all") == {
+        "committed": {
+            "added": M.unordered(
+                join("dir", "bar"),
+                join("dir", "foobar"),
+            )
+        },
+        "uncommitted": {
+            "added": [join("dir", "baz")],
```

Comment:
[Q] (probably a dumb one or could be unimportant) Is this possible? Won't using `dvc add` send a resource straight to `committed`? Is there another way to track/add an untracked resource that I'm not aware of?

Reply:
> Is this possible?

yes, for example, when you modify the dataset, but you haven't done `dvc commit` them yet.

> Won't using dvc add send a resource straight to committed?

yes, this will make them `committed`. 

> Is there another way to track/add an untracked resource that I'm not aware of?

If they are part of the dataset, eg: `dataset/file` in a `dataset`, they can be simply added as `dvc commit dataset/file`.
These will show up in `uncommitted`.

Otherwise, if they are standalone and not a part of any dvc-tracked outputs, they'll need to be `dvc add` first. They will show up as `untracked`.

Reply:
The reason I ask is I am trying to type the JSON output. Is the following correct:

```
{
  uncommitted: {
    deleted: string[]
    modified: string[]
  }
  committed: {
    added: string[]
    deleted: string[]
    modified: string[]
    renamed: string[]
  }
  not_in_cache: string[]
  unchanged: string[]
  untracked: string[]
}
```

Did I miss anything?

Reply:
That's correct, except `renamed` is not implemented yet.

Reply:
thank you 🙏🏻